### ISSUE TITLE
Failed to obtain the result of particular output stream from a particular node

### ISSUE THREAD

**Issue by YenchungChen** [google/mediapipe#1455]
I'm using MediaPipe (0.8.2) installed via pip.
Currently, I'm working with MediaPipe Holistic. Apart from following standard output stated on line 102-105 of `mediapipe/python/solutions/holistic.py`:
```python
        outputs=[
            'pose_landmarks', 'left_hand_landmarks', 'right_hand_landmarks',
            'face_landmarks'
        ]
```
I'm also interested in the output streams of other nodes in graph config `mediapipe/modules/holistic_landmark/holistic_landmark_cpu.pbtxt`, such as the debug outputs on line 72-74 of `holistic_landmark_cpu.pbtxt`:
```python
# Debug outputs
output_stream: "POSE_ROI:pose_landmarks_roi"
output_stream: "POSE_DETECTION:pose_detection"
```
However, I'm not sure what is the right way to access the results of these output streams ('pose_landmarks_roi' and 'pose_detection').
As stated in `mediapipe/python/solution_base.py`, I modified the outputs list in `holistic.py` by adding the output streams that I'm interested in, 'pose_landmarks_roi' and 'pose_detection'. After that, it seems to me that I should be able to  #obtain those output streams through something like:
```python
import mediapipe as mp

mp_holistic = mp.solutions.holistic
holistic = mp_holistic.Holistic(min_detection_confidence = 0.5)

results = holistic.process(image)
results.pose_detection
results.pose_landmarks_roi
```
And I get the following error: 
```bash
Traceback (most recent call last):
  File "camera_holistic.py", line 48, in <module>
    holistic = mp_holistic.Holistic(
  File "/home/ycchen/anaconda3/envs/tf23env/lib/python3.8/site-packages/mediapipe/python/solutions/holistic.py", line 85, in __init__
    super().__init__(
  File "/home/ycchen/anaconda3/envs/tf23env/lib/python3.8/site-packages/mediapipe/python/solution_base.py", line 219, in __init__
    canonical_graph_config_proto = self._initialize_graph_interface(
  File "/home/ycchen/anaconda3/envs/tf23env/lib/python3.8/site-packages/mediapipe/python/solution_base.py", line 356, in _initialize_graph_interface
    self._output_stream_type_info = {
  File "/home/ycchen/anaconda3/envs/tf23env/lib/python3.8/site-packages/mediapipe/python/solution_base.py", line 357, in <dictcomp>
    get_name(tag_index_name): get_stream_packet_type(tag_index_name)
  File "/home/ycchen/anaconda3/envs/tf23env/lib/python3.8/site-packages/mediapipe/python/solution_base.py", line 344, in get_stream_packet_type
    validated_graph.registered_stream_type_name(
RuntimeError: ; Unable to find the type for stream "pose_landmarks_roi".  It may be set to AnyType or something else that isn't determinable, or the type may be defined but not registered.
```
I've noticed that "validated_grah" in the error message is initialized using binarypb file (`mediapipe/modules/holistic_landmark/holistic_landmark_cpu.binarypb`). I'm also curious about the relation and the difference between `holistic_landmark_cpu.pbtxt` and `holistic_landmark_cpu.binarypb`.

Thank you for reading this!

**Comment by jiuqiant** [comment]
If you only add `pose_detection` to holistic's outputs list, it works. Then, if you do `print(results.pose_detection)` in your code, it will print out something like 

```
label_id: 0
score: 0.8247644901275635
location_data {
  format: RELATIVE_BOUNDING_BOX
  relative_bounding_box {
    xmin: 0.26724281907081604
    ymin: 0.022244099527597427
    width: 0.5384301543235779
    height: 0.5374688506126404
  }
  relative_keypoints {
    x: 0.6672111749649048
    y: 0.8083518743515015
  }
  relative_keypoints {
    x: 0.5069693326950073
    y: 0.19725963473320007
  }
  relative_keypoints {
    x: 0.5706129670143127
    y: 0.3996028006076813
  }
  relative_keypoints {
    x: 0.43811485171318054
    y: -0.11774782091379166
  }
}
```

However, adding `pose_landmarks_roi` causes the issue you mentioned above.

More specifically, `pose_landmarks_roi` can't be retrieved because the rect proto is not registered. It can be fixed by the following one line change (rebuilding the python package locally is required afterwards):

```
diff --git a/mediapipe/python/BUILD b/mediapipe/python/BUILD
index 59b20a2..8c1807a 100644
--- a/mediapipe/python/BUILD
+++ b/mediapipe/python/BUILD
@@ -47,6 +47,7 @@ pybind_extension(
         "//mediapipe/framework/formats:classification_registration",
         "//mediapipe/framework/formats:detection_registration",
         "//mediapipe/framework/formats:landmark_registration",
+       "//mediapipe/framework/formats:rect_registration",
     ],
 )
```

We can have this fix in the coming release.

**Comment by RomainSabathe** [comment]
Hi @jiuqiant, thanks for your reply. I'm facing a somewhat similar issue. My only question is: how did you know that `pose_landmarks_roi` relies on the rect proto? Where can I find that explicitly in the code? I'm getting lost in the codebase... Any help would be super appreciated. Thank you!

**Comment by jiuqiant** [comment]
> Hi @jiuqiant, thanks for your reply. I'm facing a somewhat similar issue. My only question is: how did you know that `pose_landmarks_roi` relies on the rect proto? Where can I find that explicitly in the code? I'm getting lost in the codebase... Any help would be super appreciated. Thank you!

Yes, pose_landmarks_roi is generated by https://github.com/google/mediapipe/blob/350fbb2100ad531bc110b93aaea23d96af5a5064/mediapipe/calculators/util/alignment_points_to_rects_calculator.cc and it's a NormalizedRect.

**Comment by jiuqiant** [comment]
The fix is releases in v0.8.3. Closing this issue now.
